src/formatDate.js

```diff
@@ -8,7 +8,49 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const formattedToDateParts = {}; // day, month, year
+  const values = date.split(fromFormat[3]);
+  let formattedDate = '';
+
+  for (let i = 0; i < 3; i++) {
+    if (fromFormat[i] === 'YYYY' && toFormat[i] === 'YY') {
+      formattedToDateParts[2] = values[i].slice(-2);
+    } else if (fromFormat[i] === 'YY' && toFormat[i] === 'YYYY') {
+      const year = Number(values[i]);
+
+      if (year < 30) {
+        formattedToDateParts[2] = '20' + values[i];
+      } else {
+        formattedToDateParts[2] = '19' + values[i];
+      }
+    } else if (fromFormat[i] === 'YY' || fromFormat[i] === 'YYYY') {
+      formattedToDateParts[2] = values[i];
+    }
+
+    if (fromFormat[i] === 'DD') {
+      formattedToDateParts[0] = values[i];
+    }
+
+    if (fromFormat[i] === 'MM') {
+      formattedToDateParts[1] = values[i];
```

Comment:
The assignment `formattedToDateParts[1]` assumes that the month will always be placed at the second position in the `toFormat` array. You should dynamically determine the correct index to use based on the `toFormat`.